menelaus/injection/injector.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABC, abstractclassmethod
+
+import numpy as np
+import pandas as pd
+
+
+class Injector(ABC):
+    """
+    Abstract base class for drift injection callables. Classes using this
+    pattern should implement a ``__call__`` function, and also make use of the
+    ``_preprocess()`` and ``_postprocess`` type handling capabilities of this
+    class.
+
+    When called, an ``Injector`` typically accepts either ``pandas.DataFrame``
+    or ``numpy.ndarray`` data, along with other parameters for the injection
+    technique it is modeling, and returns the data with drift injected over a
+    specified window.
+    """
+
+    def _preprocess(self, data, *columns, return_df=False):
```

Comment:
@Anmol-Srivastava Code looks great -  out of curiosity, what circumstances could you see a user specifying return_df = T, since the injection methods use np.arrays? Is that option built in mostly for flexibility of the injector class?

Reply:
Basically `FeatureCoverInjector` (formerly `feature_hide_and_sample`) needs this to group-by the dropped feature before resampling from each group. After that I thought I would extend the option to all future injectors by putting it in the parent class.